### DISCUSSION TITLE
Is there a way to have multiple actions on the "on_click" function?

### DISCUSSION THREAD

**Discussion by chrsBlank** [zauberzeug/nicegui#735] [Q&A] [answered]
### Question

I have this piece of code
```py
with ui.row().classes('w-full items-center'):
    result = ui.label().classes('mr-auto')
    with ui.button(on_click=lambda: menu.open()).props('icon=menu'):
        with ui.menu() as menu:
            ui.menu_item('button1l',lambda: button1())
            ui.menu_item('button2',lambda: button2())
```

and i would like to have the function resetAction(label1, label2, num1) when i open the menu

is it possible? i overcame this by having a function in the on click but its not possible on the init of the menu : ((

**Comment by chrsBlank** [comment]
never mind, i tried to initialize the initialization of the menu in another function and it worked. Though if anyone knows how to stack functions in the on_click one it would be very helpful

**Comment by falkoschindler** [comment, marked as answer]
Sure. The "normal" way to combine multiple actions would be to define a dedicated function for handling the button click:

```py
def action1():
    ui.notify('action1')

def action2():
    ui.notify('action2')

def handle_click():
    action1()
    action2()

ui.button(on_click=handle_click)
```

A sneaky shortcut is to combine both calls in a lambda statement using a tuple (or list):

```py
ui.button(on_click=lambda: (action1(), action2()))
```

**Reply by chrsBlank** [reply]
oh wow, i shall try this, thank you
